dfastmi/batch/FileNameRetriever.py

```diff
@@ -29,119 +29,195 @@
 
 from typing import Optional, Union, Dict, Any, Tuple
 import configparser
+from abc import ABC, abstractmethod
+from packaging import version
 
-def get_filenames_version1(
-    config: Optional[configparser.ConfigParser] = None,
-) -> Dict[Any, Tuple[str,str]]:
+class IFileNameRetriever(ABC):
```

Comment:
Why IFileNameRetriever and not AFileNameRetriever, there is a protected method in this class..

Reply:
todo

Reply:
how do we do protected methods?

Reply:
done, question in regard to protected method still stands.